workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs

```diff
@@ -12,38 +12,39 @@ namespace Improbable.Gdk.GameObjectCreation
     ///     and links it to the entity via the EntityGameObjectLinker. Also checks for entity removal and calls the
     ///     IEntityGameObjectCreator for cleanup.
     /// </summary>
-    [UpdateInGroup(typeof(SpatialOSReceiveGroup.EntityInitialisationGroup))]
     [DisableAutoCreation]
     internal class GameObjectInitializationSystem : ComponentSystem
     {
-        // TODO: Add our ISystemStateComponentData for all entities instead of ones with GameObjects, store spatial ID
-        // callbacks instead of in it, give that in removal callback instead of SpatialOSEntity struct.
+        public struct HasGameObjectSystemState : ISystemStateComponentData
```

Comment:
Minor (may help with the other comment too) - if it has fields in it, it could be renamed to something else, e.g. `SpatialEntityIdData` or something... - can't think of good names here but I am guessing you get my point)

Reply:
In this case the name should represent what the component denotes. The data is just what is needed to clean up. Putting this in an injection group is meant to denote that is has a game object associated with it.